From 94691560a99a75946087f11968369179b81e194a Mon Sep 17 00:00:00 2001
From: Selfmuser <109387456+selfmusing@users.noreply.github.com>
Date: Sun, 24 Aug 2025 16:04:27 +0530
Subject: [PATCH 04/04] BACKPORT: seccomp: add filter_count field for KernelSU

KernelSU requires struct seccomp to have atomic_t filter_count ,
but this field is missing on NON-GKI kernels.

Adding the field fixes build errors such as:
error: no member named 'filter_count' in 'struct seccomp'

Reference: https://github.com/tiann/KernelSU/pull/2708

Change-Id: Ief7567bfdf749d3cd8c67e9fbdcea15f956a4f05
Signed-off-by: Neebe3289 <neebexd@gmail.com>
---
 include/linux/seccomp.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/seccomp.h b/include/linux/seccomp.h
index a9d5c52de4ea..03c17f0a0e6b 100644
--- a/include/linux/seccomp.h
+++ b/include/linux/seccomp.h
@@ -12,6 +12,7 @@
 
 #include <linux/thread_info.h>
 #include <asm/seccomp.h>
+#include <linux/atomic.h>
 
 struct seccomp_filter;
 /**
@@ -28,6 +29,7 @@ struct seccomp_filter;
 struct seccomp {
 	int mode;
 	struct seccomp_filter *filter;
+	atomic_t filter_count;
 };
 
 #ifdef CONFIG_HAVE_ARCH_SECCOMP_FILTER
-- 
2.51.2

